gadget/device_darwin.go

```diff
@@ -25,6 +25,8 @@ import (
 	"github.com/snapcore/snapd/gadget/quantity"
 )
 
+var ErrDeviceNotFound = errors.New("device not found")
```

Comment:
this should live elsewhere I think instead of having a version for _darwin and one for _linux

Reply:
I moved some things around for this